bin/clinvar_jsons/parse_trait_names.py

```diff
@@ -56,6 +76,36 @@ def clinvar_jsons(filepath):
             yield json.loads(line)
 
 
+class Trait:
+    def __init__(self, name):
+        self.name = name
+        self.xref_set = set()
+        self.count = 1
+
+    # def __eq__(self, other):
+    #     if isinstance(other, Trait):
+    #         return self.name == other.name and set(self.xref_set) == set(other.xref_set)
+    #     else:
+    #         return False
+    #
+    # def __hash__(self):
+    #     return hash((self.name, self.xref_set))
+    #
+    # def __ne__(self, other):
+    #     return not (self == other)
+
+    @property
+    def xref_string(self):
+        return "|".join(["{}/{}".format(xref.db, xref.id_)
+                         for xref in self.xref_set if xref.status.lower() == "current"])
+
+    def __str__(self):
+        return "{}\t{}\t{}".format(self.name, self.xref_string, self.count)
+
+
+TraitXref = namedtuple("TraitXref", ["db", "id_", "status"])
```

Comment:
This can essentially be treated as a very basic class (https://docs.python.org/3/library/collections.html#collections.namedtuple). A object is created using this namedtuple here https://github.com/tomdcsmith/eva-cttv-pipeline/blob/clinvar_trait_hints/bin/clinvar_jsons/parse_trait_names.py#L65 and then the attributes (db, id_ and status) can then be referenced just like a regular class https://github.com/tomdcsmith/eva-cttv-pipeline/blob/clinvar_trait_hints/bin/clinvar_jsons/parse_trait_names.py#L100. It's just a lot more lightweight and required much less code.